### PR TITLE
[CPROD-1163] Update IC dependency

### PR DESCRIPTION
**Pull request by awkure**
Currently there is a bug when trying to deploy v1-amm:

```
Error: The Replica returned an error: code 5, message: "Wasm module of canister rrkah-fqaaa-aaaaa-aaaaq-cai is not valid: Wasm module has an invalid import section. Module imports function '__wbindgen_describe' from '__wbindgen_placeholder__' that is not exported by the runtime."
``` 

That is because ic devs have inconsistent versioning in their codebase. We've decided to upgrade the ic dependencies to be able to use their canisters further.

### CODE REVIEW

File: ic-factory/src/api.rs [hunk 1]
```diff
@@ -26,6 +26,7 @@ pub trait FactoryCanister: Canister + Sized + PreUpdate {
     /// otherwise, cycles balances of `principal` is returned.
     /// If `principal` does not exists, `None` is returned.
     #[update(trait = true)]
+    #[allow(clippy::needless_lifetimes)]
     fn get_cycles<'a>(&'a self, principal: Option<Principal>) -> AsyncReturn<Option<Nat>> {
```
Comment (hagsteel) [inline ic-factory/src/api.rs:30]: I thought this was solved, the issue with the life times.
This should be valid to write as 
```rust
fn get_cycles(&self, principal: Option<Principal>) -> AsyncReturn<Option<Nat>>
```
and if `AsyncReturn` has a life time argument it should be written as `AsyncReturn<'_, Option<Nat>>`

Was this never fixed or do we have a regression?

File: ic-helpers/src/management/canister.rs [hunk 2]
```diff
@@ -443,13 +441,16 @@ pub struct CallSignature {
 pub fn der_encode_pub_key(pk: &Pubkey) -> Vec<u8> {
     let pubkey = PublicKey::parse_slice(pk.as_bytes(), None).expect("not a valid public key");
     let pubkey_bytes_uncompress = pubkey.serialize();
-    let der_encoded_public_key: PublicKeyDocument = SubjectPublicKeyInfo {
-        algorithm: Secp256k1::algorithm_identifier(),
+    let der_encoded_public_key: pkcs8::Document = SubjectPublicKeyInfo {
+        algorithm: AlgorithmIdentifier {
+            oid: ObjectIdentifier::new("1.2.840.10045.2.1")
```
Comment (hagsteel) [inline ic-helpers/src/management/canister.rs:446]: Don't use `expect` here. If we are hell bent on doing this at least use the supplied function `ObjectIdentifier::new_unwrap`

**Review by hagsteel — APPROVED** [review]
Approving but please review the comments (the suggestions is just changing error messages to lower case for consistency)

File: ic-factory/src/api.rs (hunk 1 above)
Comment (awkure) [reply ic-factory/src/api.rs:30]: 🤷

Comment (hagsteel) [reply ic-factory/src/api.rs]: Alright, leave it for now, we'll make a separate issue for this

File: ic-canister/ic-canister-macros/src/api.rs [hunk 3]
```diff
@@ -102,19 +102,11 @@ pub(crate) fn api_method(
     let mut args_destr = Punctuated::new();
     let mut has_self = false;
 
-    let mut self_lifetime = quote! {};
-
     for arg in args {
         let (arg_type, arg_pat) = match arg {
             FnArg::Receiver(r) => {
```
Comment (itsyaasir) [inline ic-canister/ic-canister-macros/src/api.rs:107]:
```suggestion
            FnArg::Receiver(_) => {
```

**Review by itsyaasir — APPROVED** [review]
Everything looks good :)

**Review by F3kilo — APPROVED** [review]
We can remove one `expect`, but I think it is not necessary.

**Pull request merged**